src/components/OfflineWithFeedback.tsx

```diff
@@ -0,0 +1,136 @@
+import lodashOmitBy from 'lodash/omitBy';
+import React from 'react';
+import {ImageStyle, StyleProp, TextStyle, View, ViewStyle} from 'react-native';
+import useNetwork from '@hooks/useNetwork';
+import shouldRenderOffscreen from '@libs/shouldRenderOffscreen';
+import styles from '@styles/styles';
+import * as StyleUtils from '@styles/StyleUtils';
+import CONST from '@src/CONST';
+import * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import MessagesRow from './MessagesRow';
+
+/**
+ * This component should be used when we are using the offline pattern B (offline with feedback).
+ * You should enclose any element that should have feedback that the action was taken offline and it will take
+ * care of adding the appropriate styles for pending actions and displaying the dismissible error.
+ */
+
+type OfflineWithFeedbackProps = {
+    /** The type of action that's pending  */
+    pendingAction: OnyxCommon.PendingAction | null;
+
+    /** Determine whether to hide the component's children if deletion is pending */
+    shouldHideOnDelete?: boolean;
+
+    /** The errors to display  */
+    errors?: OnyxCommon.Errors | null;
+
+    /** Whether we should show the error messages */
+    shouldShowErrorMessages?: boolean;
+
+    /** Whether we should disable opacity */
+    shouldDisableOpacity?: boolean;
+
+    /** A function to run when the X button next to the error is clicked */
+    onClose?: () => void;
+
+    /** The content that needs offline feedback */
+    children: React.ReactNode;
+
+    /** Additional styles to add after local styles. Applied to the parent container */
+    style?: StyleProp<ViewStyle>;
+
+    /** Additional styles to add after local styles. Applied to the children wrapper container */
+    contentContainerStyle?: StyleProp<ViewStyle>;
+
+    /** Additional style object for the error row */
+    errorRowStyles?: StyleProp<ViewStyle>;
+
+    /** Whether applying strikethrough to the children should be disabled */
+    shouldDisableStrikeThrough?: boolean;
+
+    /** Whether to apply needsOffscreenAlphaCompositing prop to the children */
+    needsOffscreenAlphaCompositing?: boolean;
+
+    /** Whether we can dismiss the error message */
+    canDismissError?: boolean;
+};
+
+/**
+ * This method applies the strikethrough to all the children passed recursively
+ */
+function applyStrikeThrough(children: React.ReactNode) {
+    return React.Children.map(children, (child) => {
+        if (!React.isValidElement(child)) {
+            return child;
+        }
+        const props: {style: Array<ViewStyle | TextStyle | ImageStyle>; children?: React.ReactNode} = {
```

Comment:
Create separate type for this `{style: Array<ViewStyle | TextStyle | ImageStyle>; children?: React.ReactNode}`